src/main/lens-proxy.ts

```diff
@@ -185,15 +185,25 @@ export class LensProxy extends Singleton {
   }
 
   protected async handleRequest(req: http.IncomingMessage, res: http.ServerResponse) {
+    // allow to fetch apis in "clusterId.localhost:port" from "localhost:port"
+    // this should be safe because we have already validated cluster uuid
+        
+    res.setHeader("Access-Control-Allow-Origin", "*");
+    res.setHeader("Access-Control-Allow-Methods", "GET, POST, PUT, DELETE, PATCH, OPTIONS");
+    res.setHeader("Access-Control-Allow-Headers", "Content-Type, X-Cluster-Id");
+
+    if (req.method === "OPTIONS") {
+      res.end();
```

Comment:
What the point to allow `OPTIONS` from `Access-Control-Allow-Methods` to end it right after?

Reply:
Renderer (browser) asks CORS via `OPTIONS`, we just need to return headers and nothing else.